YOLayout/YOLayout_IMP.m

```diff
@@ -142,24 +149,21 @@ - (CGRect)setFrame:(CGRect)frame inRect:(CGRect)inRect view:(id)view options:(YO
   if ((options & YOLayoutOptionsSizeToFitVertical) == 0) {
     frame.size.height = originalFrame.size.height;
   }
-  
-  CGRect rect = originalFrame;
-  if (!CGRectIsEmpty(inRect)) rect = inRect;
-  
-  if ((options & YOLayoutOptionsAlignCenterVertical) != 0) {
-    frame = YOCGRectToCenterYInRect(frame, originalFrame);
-  }
 
-  if ((options & YOLayoutOptionsAlignCenterHorizontal) != 0) {
-    frame = YOCGRectToCenterXInRect(frame, originalFrame);
+  // If we passed in 0 for inRect height, then lets set it to the frame height.
+  // This usually happens if we are sizing to fit, and is needed to align below.
+  if (inRect.size.height == 0) inRect.size.height = frame.size.height;
+  
+  if ((options & YOLayoutOptionsAlignCenter) != 0) {
+    frame = YOCGRectToCenterInRect(frame, inRect);
   }
 
   if ((options & YOLayoutOptionsAlignRight) != 0) {
-    frame = YOCGRectRightAlignWithRect(frame, rect);
+    frame = YOCGRectRightAlignWithRect(frame, inRect);
```

Comment:
I don't think I understand this. Seems like this change would break right align since `inRect` is `CGRectZero` when using `setFrame:view:options`.